lib/imsg-buffer.c

```diff
@@ -209,8 +209,10 @@ msgbuf_clear(struct msgbuf *msgbuf)
 {
 	struct ibuf	*buf;
 
-	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL)
+	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL) {
 		ibuf_dequeue(msgbuf, buf);
+		TAILQ_FIRST(&msgbuf->bufs) = NULL;
```

Comment:
@donaldsharp that should fix the warning, but there's nothing wrong with the current code. Unfortunately clang's static analyzer is not smart enough to see that TAILQ_FIRST() always returns different pointers in this case, as ibuf_dequeue() calls TAILQ_REMOVE() to remove the first element of the list.

I'd prefer to just ignore this warning and keep our imsg files identical to the original ones from OpenBSD.